AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -407,6 +411,33 @@ - (void)setLogUrl:(NSString *)logUrl {
   }
 }
 
+- (void)setMaxStorageSize:(long)sizeInBytes completionHandler:(void (^)(BOOL))completionHandler {
```

Comment:
I think it makes sense to prevent small values < 20KiB at this point instead of later.

Reply:
Would you mind elaborating on that? The original rationale was that since the storage limitation is on the Storage side of things, which app center is aware of, the separation of responsibility is most clear if that limit is enforced by `MSLogDbStorage` class (there shouldn't be a noticeable performance difference either way).

Not opposed to the change if there's a good reason, just wondering though.

Reply:
The 20KiB size check is like parameter validation of the `setMaxStorageSize` API, so it make sense to have the validation logic here instead of further down in database class. I agree with @jwallra on this.
Android implementation does 20KiB check in the `setMaxStorageSize` implementation, though it's fine to have slightly different implementation on Android/iOS.

Reply:
@achocron, @dhei basically has the crux of it, checking here is basically parameter validation, and we can improve performance because we already know here whether or not we can set the given size because it's part of restrictions of the API itself.